### ISSUE TITLE
Should constant be named fill

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaComputing/ArrayFire.jl#138]
... or am I missing some difference in behavior? ;)

**Comment by SimonDanisch** [comment]
Of course besides that it returns an AFArray. That means it would be mandatory to pass the array type like for `ones`.

**Comment by ranjanan** [comment]
Hi @SimonDanisch, apologies for not getting to this earlier, but the problem I face is this: 

```julia
julia> import Base.fill

julia> fill{T}(::Type{AFArray{T}}, dims) = constant(T, dims...)
fill (generic function with 3 methods)

julia> fill(AFArray{Float32}, 2, 2)
2×2 Array{DataType,2}:
 ArrayFire.AFArray{Float32,N}  ArrayFire.AFArray{Float32,N}
 ArrayFire.AFArray{Float32,N}  ArrayFire.AFArray{Float32,N}
```
I guess this is because fill takes ANY in the first field. Please let me know if you have a workaround :)

**Comment by SimonDanisch** [comment]
I think you didn't actually overload the function you're calling, since you forgot `...` after dims!
This works as expected:
```Julia
julia> immutable Test{T, N} end

julia> function Base.fill{T, N}(::Type{Test{T, N}}, dims::Integer...) 
       return Test{T, N}()
       end

julia> fill(Test{Float32, 2}, 2, 2)
Test{Float32,2}()

julia> function Base.fill{T}(::Type{Test{T}}, dims::Integer...) 
       return Test{T, 77}()
       end

julia> fill(Test{Float32}, 2, 2)
Test{Float32,77}()
```

**Comment by gaika** [comment]
it is `fill` in #121

**Comment by gaika** [comment]
merged with [v0.1.0](https://github.com/JuliaComputing/ArrayFire.jl/releases/tag/v0.1.0)
